server/src/controllers/job_posts/getAllJobPosts.js

```diff
@@ -0,0 +1,85 @@
+// Get All Job Posts
+// a) GET request to get all job posts [id, title, employer, application_close_date, state]
+// Return the id, title, employer, application_close_date, state values of ALL job posts
+
+// b) GET request to get all public job posts 
+// Return the id, title, employer, location, application_close_date of job posts where state is "Active"
+
+// c) GET request by job type using job type parameter
+// Return all job posts with that job type and state = "Active"
+
+const { query } = require("express");
+const JobPosting = require("../../models/job_posting.model");
+const logger = require("pino")();
+
+
+const getAllJobPostsRequestHandler = async (req, res) => {
+    // check method is GET
+    if (req.method !== 'GET') {
+        return res.status(405).json({ message: 'Method not allowed, only GET methods allowed.' });
+    }
+
+    try{
+        // ------ Pagination Configs
+        const searchConfig = {
+            where: query,
+            attributes,
+            limit,
+            offset
+        };
+
+        // get pagination parameters:
+        const page = req?.query?.page ? parseInt(req.query.page, 10) : null;
+        const pageSize = req?.query?.pageSize
+            ? parseInt(req.query.pageSize, 10)
+            : null;
+
+        if (page != null && pageSize != null) {
+            searchConfig.limit = pageSize;
+            searchConfig.offset = page * pageSize;
+        }
+        // --------
+
+        // Return to include id, title, employer, application_close_date, state
+        searchConfig.attributes = ['id', 'title', 'employer', 'close_date', 'state'];
+
+        // a) Get all Job Posts
+        const allJobPosts = await JobPosting.findAndCountAll(searchConfig);
+        
+        // b) Get all Public Job Posts
+        query.state = 'Active';
+        const allActiveJobPosts = await JobPosting.findAndCountAll(searchConfig);
+
+        // c)
+        query.job_type = req?.query?.job_type
+        const allActiveJobPostsByType = await JobPosting.findAndCountAll(searchConfig);
```

Comment:
Doesn't seem like allActiveJobPostsByType is included anywhere in the response object - If this query is supposed to be returned in the response object that is.